files/default/kagent_utils/kagent_utils/monitoring/service.py

```diff
@@ -129,7 +140,7 @@ def _exec_check_call(self, command, stdout=None, stderr=None):
         subprocess.check_call(command, stdout=stdout, stderr=stderr)
 
     def _exec_check_output(self, command, stderr=None):
-        subprocess.check_output(command, stderr=stderr)
+        return subprocess.check_output(command, stderr=stderr).strip()
```

Comment:
don't you need to decode it, i remember we were doing that in ec2-init

Reply:
No, the `check_output` does it for you. In ec2-init we use `subprocess.run()`